### PR TITLE
Features/uintra2 16 move groups

### PR DESCRIPTION
**Pull request by eugene-sahach**

### CODE REVIEW

File: src/Uintra20/Content/uintra20/src/app/feature/project/reusable/ui-elements/publication-header/publication-header.component.html [hunk 1]
```diff
@@ -15,8 +15,15 @@
       {{ title }}
     </a>
     <ng-container *ngIf='false'>{{ title }}</ng-container>
-    <div class="publication-header__subtitle">
+    <div class="publication-header__subtitle-container">
       <ng-content select="[subtitle]"></ng-content>
+      <span *ngIf="groupInfo" class="publication-header__group">
+        >
+        <a [routerLink]="groupInfo.url?.originalUrl" class="publication-header__group-link">
```
Comment (razrabotal) [inline src/Uintra20/Content/uintra20/src/app/feature/project/reusable/ui-elements/publication-header/publication-header.component.html]: please add pipe | ulink

File: src/Uintra20/Content/uintra20/src/app/feature/project/reusable/ui-elements/publication-header/publication-header.component.ts [hunk 2]
```diff
@@ -1,4 +1,5 @@
 import { Component, Input, OnInit } from '@angular/core';
+import { IUlinkWithTitle, IULink } from 'src/app/feature/shared/interfaces/IULink';
```
Comment (razrabotal) [inline src/Uintra20/Content/uintra20/src/app/feature/project/reusable/ui-elements/publication-header/publication-header.component.ts]: IUlinkWithTitle - why is this needed?

File: src/Uintra20/Content/uintra20/src/app/feature/project/specific/activity/activity.interfaces.ts [hunk 3]
```diff
@@ -40,6 +41,7 @@ export interface ISocialEdit {
   availableTags: Array<ITagData>;
   lightboxPreviewModel: ILightBoxPreviewModel;
   id: string;
+  groupId: string;
```
Comment (razrabotal) [inline src/Uintra20/Content/uintra20/src/app/feature/project/specific/activity/activity.interfaces.ts]: Maybe `groupId?: string;` ?

File: src/Uintra20/Content/uintra20/src/app/feature/project/specific/groups/groups-form/groups-form.component.ts [hunk 4]
```diff
@@ -71,14 +74,18 @@ export class GroupsFormComponent {
       if (!this.edit) {
         this.groupsService.createGroup(groupModel).pipe(
           finalize(() => this.inProgress = false)
-        ).subscribe(res => {});
+        ).subscribe(res => {
+          this.router.navigate([res.originalUrl]);
```
Comment (razrabotal) [inline src/Uintra20/Content/uintra20/src/app/feature/project/specific/groups/groups-form/groups-form.component.ts]: add pipe ulink

File: src/Uintra20/Content/uintra20/src/app/feature/project/specific/groups/groups-form/groups-form.component.ts [hunk 5]
```diff
@@ -71,14 +74,18 @@ export class GroupsFormComponent {
...
+        });
       } else {
         if (this.data && this.data.media && this.data.media.length) {
           groupModel.media = this.medias[0];
         }
         this.groupsService.editGroup(groupModel).pipe(
           finalize(() => this.inProgress = false)
-        ).subscribe(res => {});
+        ).subscribe(res => {
+          this.router.navigate([res.originalUrl]);
```
Comment (razrabotal) [inline src/Uintra20/Content/uintra20/src/app/feature/project/specific/groups/groups-form/groups-form.component.ts]: add pipe ulink

File: src/Uintra20/Content/uintra20/src/app/feature/project/specific/groups/groups.service.ts [hunk 6]
```diff
@@ -16,17 +16,17 @@ export interface IGroupModel {
...
+//   id: string;
+//   title: string;
+//   description: string;
+//   createdDate: string;
+//   updatedDate: string;
+//   creatorId: string;
+//   imageId?: string;
+//   isHidden: boolean;
+//   groupTypeId: number;
+// }
```
Comment (razrabotal) [inline src/Uintra20/Content/uintra20/src/app/feature/project/specific/groups/groups.service.ts]: remove unused code

File: src/Uintra20/Content/uintra20/src/app/ui/pages/news/uintra-news-details/uintra-news-details-page.component.ts [hunk 7]
```diff
@@ -35,6 +35,7 @@ export class UintraNewsDetailsPage implements OnInit {
 
   public ngOnInit(): void {
     this.parsedData = ParseHelper.parseUbaselineData(this.data);
+    console.log(this.parsedData)
```
Comment (razrabotal) [inline src/Uintra20/Content/uintra20/src/app/ui/pages/news/uintra-news-details/uintra-news-details-page.component.ts]: remove it

File: src/Uintra20/Content/uintra20/src/app/ui/pages/news/uintra-news-details/uintra-news-details-page.html [hunk 8]
```diff
@@ -1,85 +1,97 @@
...
+
+            <div class="feed-item__title" [innerHtml]="detailsTitle"></div>
+            <div class="feed-item__text rte" [innerHtml]="detailsDescription"></div>
+            <div class="feed-item-tags" *ngIf="tags.length !== 0">
+              <span class="feed-item-tags__text">This activity is tagged</span>
+              <!-- TODO: get from details.links -->
+              <a
+                *ngFor="let tag of tags"
+                class="feed-item-tags__tag"
+                [routerLink]="['/search']"
```
Comment (razrabotal) [inline src/Uintra20/Content/uintra20/src/app/ui/pages/news/uintra-news-details/uintra-news-details-page.html]: fix hardcoded link

File: src/Uintra20/Content/uintra20/src/app/ui/pages/social/details/social-details-page.component.html [hunk 9]
```diff
@@ -1,78 +1,90 @@
...
+            <a [routerLink]="['/']" class="feed-header__back-link">Go to List</a>
             <!-- TODO: get from details.links -->
             <a
-              *ngFor="let tag of tags"
-              class="feed-item-tags__tag"
-              [routerLink]="['/search']"
-              [queryParams]="{ query: tag.text }"
-              [innerHTML]="tag.text"
-            ></a>
+              [routerLink]="['/social-edit']"
```
Comment (razrabotal) [inline src/Uintra20/Content/uintra20/src/app/ui/pages/social/details/social-details-page.component.html]: fix hardcoded link

File: src/Uintra20/Content/uintra20/src/app/ui/pages/social/details/social-details-page.component.html [hunk 10]
```diff
@@ -1,78 +1,90 @@
...
+              [activityName]="activityName"
+            ></app-detailas-header>
+            <div class="feed-item__text rte" [innerHtml]="detailsDescription"></div>
+            <div class="feed-item-tags" *ngIf="tags.length !== 0">
+              <span class="feed-item-tags__text">This activity is tagged</span>
+              <!-- TODO: get from details.links -->
+              <a
+                *ngFor="let tag of tags"
+                class="feed-item-tags__tag"
+                [routerLink]="['/search']"
```
Comment (razrabotal) [inline src/Uintra20/Content/uintra20/src/app/ui/pages/social/details/social-details-page.component.html]: fix hardcoded link

File: src/Uintra20/Content/uintra20/src/app/ui/pages/social/details/social-details-page.component.html [hunk 11]
```diff
@@ -1,78 +1,90 @@
...
+                      class="documents-list__link"
+                      [href]="document.url"
+                      target="_blank"
+                    >
+                      <span class="documents-list__icon file-extension">{{
+                        document.extension
+                      }}</span>
+                      <span class="documents-list__title">{{ document.name }}</span>
+                    </a>
+                  </li>
```
Comment (razrabotal) [inline src/Uintra20/Content/uintra20/src/app/ui/pages/social/details/social-details-page.component.html]: maybe move to reusable component

File: src/Uintra20/Content/uintra20/src/app/ui/pages/social/edit/social-edit-page.component.html [hunk 12]
```diff
@@ -1,59 +1,73 @@
...
+          <button class="button button-default"
+            [disabled]="inProgress"
+            (click)="handleSocialDelete()">DELETE</button>
 
-      <a class="button button-default"
-        [routerLink]="['/social-details']"
-        [queryParams]="{id: socialEdit.id}">CANCEL</a>
-    </div>
+          <a class="button button-default"
+            [routerLink]="['/social-details']"
```
Comment (razrabotal) [inline src/Uintra20/Content/uintra20/src/app/ui/pages/social/edit/social-edit-page.component.html]: fix hardcoded link

File: src/Uintra20/Content/uintra20/src/app/ui/pages/uintra-groups/room/uintra-groups-room-page.html [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+              [innerHTML]="parsedData.groupInfo?.description">
+            </div>
+            <span *ngIf="parsedData?.groupInfo?.membersCount">
+              {{parsedData.groupInfo.membersCount}} {{parsedData.groupInfo.membersCount ? 'member' : 'members'}} -
+            </span>
+            <span *ngIf="parsedData?.groupInfo?.creator">
+              Owner:
+              <a
+                class="right-column__info-owner-link"
+                [routerLink]="['/profile']"
```
Comment (razrabotal) [inline src/Uintra20/Content/uintra20/src/app/ui/pages/uintra-groups/room/uintra-groups-room-page.html]: fix hardcoded link

File: src/Uintra20/Content/uintra20/src/app/ui/panels/activity-create/sections/news-create/news-create.component.ts [hunk 14]
```diff
@@ -36,16 +36,17 @@ export class NewsCreateComponent implements OnInit {
...
-      .subscribe((r: any) => {
-        this.routerResolverService.removePageRouter(r.originalUrl);
-        this.router.navigate([r.originalUrl]);
-      });
+    if (this.panelData.groupId) {data.groupId = this.panelData.groupId}
+    this.activityService
+    .submitNewsContent(data)
+    .subscribe((r: any) => {
+      this.routerResolverService.removePageRouter(r.originalUrl);
+      this.router.navigate([r.originalUrl]);
```
Comment (razrabotal) [inline src/Uintra20/Content/uintra20/src/app/ui/panels/activity-create/sections/news-create/news-create.component.ts]: add pipe ulink

File: src/Uintra20/Content/uintra20/src/app/feature/project/reusable/ui-elements/publication-header/publication-header.component.html (hunk 1 above)
Comment (eugene-sahach) [reply src/Uintra20/Content/uintra20/src/app/feature/project/reusable/ui-elements/publication-header/publication-header.component.html]: +

File: src/Uintra20/Content/uintra20/src/app/feature/project/reusable/ui-elements/publication-header/publication-header.component.ts (hunk 2 above)
Comment (eugene-sahach) [reply src/Uintra20/Content/uintra20/src/app/feature/project/reusable/ui-elements/publication-header/publication-header.component.ts]: +

File: src/Uintra20/Content/uintra20/src/app/feature/project/specific/activity/activity.interfaces.ts (hunk 3 above)
Comment (eugene-sahach) [reply src/Uintra20/Content/uintra20/src/app/feature/project/specific/activity/activity.interfaces.ts]: +

File: src/Uintra20/Content/uintra20/src/app/feature/project/specific/groups/groups.service.ts (hunk 6 above)
Comment (eugene-sahach) [reply src/Uintra20/Content/uintra20/src/app/feature/project/specific/groups/groups.service.ts]: +

File: src/Uintra20/Content/uintra20/src/app/ui/pages/news/uintra-news-details/uintra-news-details-page.component.ts (hunk 7 above)
Comment (eugene-sahach) [reply src/Uintra20/Content/uintra20/src/app/ui/pages/news/uintra-news-details/uintra-news-details-page.component.ts]: +

File: src/Uintra20/Content/uintra20/src/app/ui/pages/social/details/social-details-page.component.html (hunk 10 above)
Comment (eugene-sahach) [reply src/Uintra20/Content/uintra20/src/app/ui/pages/social/details/social-details-page.component.html]: +

File: src/Uintra20/Content/uintra20/src/app/ui/pages/news/uintra-news-details/uintra-news-details-page.html (hunk 8 above)
Comment (eugene-sahach) [reply src/Uintra20/Content/uintra20/src/app/ui/pages/news/uintra-news-details/uintra-news-details-page.html]: +

File: src/Uintra20/Content/uintra20/src/app/ui/pages/social/details/social-details-page.component.html (hunk 9 above)
Comment (eugene-sahach) [reply src/Uintra20/Content/uintra20/src/app/ui/pages/social/details/social-details-page.component.html]: +
